src/lib/nvme_pt_ioctl.c

```diff
@@ -2311,3 +2320,70 @@ int opal_write_shadow_mbr_pt(struct sed_device *dev, const struct sed_key *key,
 	opal_end_session(dev->fd, opal_dev);
 	return ret;
 }
+
+
+int opal_revertsp_pt(struct sed_device * dev, const struct sed_key *key, bool keep_key)
```

Comment:
We already have this implemented.
You can use it.

Reply:
if you are referring to the opal_revertsp_pt function. If we already have this implemented, does that mean you want me to remove the opal_revertsp_pt ? If that is the case, and the other function I wrote  opal_non_destructive_revert_pt is "repetitive" that means that everything I wrote is not needed. Im confused.

Reply:
Yes opal_revertsp_pt() function duplicates opal_revertlsp_pt() function. Also opal_non_destructive_revert_pt() is unnecesary. In opal_revertper_pt() function you can do following:

```
int opal_reverttper_pt(struct sed_device *dev, const struct sed_key *key,
		bool psid, bool non_destructive)
{
......
if (non_destructive){
	ret = opal_revertlsp_pt(dev, key, true);
        if (ret) 
            return ret;
}
....
}
```
This will basically achieve the same effect as your 2 new functions